apps/discussions/contracts/DiscussionApp.sol

```diff
@@ -0,0 +1,125 @@
+pragma solidity ^0.4.24;
+
+import "@aragon/os/contracts/apps/AragonApp.sol";
+import "@aragon/os/contracts/lib/math/SafeMath.sol";
+import "@aragon/os/contracts/common/IForwarder.sol";
+
+
+contract DiscussionApp is IForwarder, AragonApp {
+    using SafeMath for uint256;
+
+    event Post(address indexed author, string postCid, uint discussionThreadId, uint postId, uint createdAt);
+    event Revise(address indexed author, string revisedPostCid, uint discussionThreadId, uint postId, uint createdAt, uint revisedAt);
+    event Hide(address indexed author, uint discussionThreadId, uint postId, uint hiddenAt);
+    event CreateDiscussionThread(uint actionId, bytes _evmScript);
+
+    bytes32 constant public DISCUSSION_POSTER_ROLE = keccak256("DISCUSSION_POSTER_ROLE");
+    string private constant ERROR_CAN_NOT_FORWARD = "DISCUSSIONS_CAN_NOT_FORWARD";
+
+    struct DiscussionPost {
+        address author;
+        string postCid;
+        uint discussionThreadId;
+        uint id;
+        uint createdAt;
+        bool show;
+        string[] revisionCids;
+    }
+
+    uint discussionThreadId;
+
+    mapping(uint => DiscussionPost[]) public discussionThreadPosts;
```

Comment:
Not sure how this will behave on an upgrade... this may need to be a mapping into another mapping instead of an array.